src/setupWorker/setupWorker.ts

```diff
@@ -49,6 +49,11 @@ export function setupWorker(
           navigator.serviceWorker,
           'message',
           (event: MessageEvent) => {
+            // Avoid messages broadcasted from unrelated workers.
+            if (event.source !== context.worker) {
```

Comment:
I think this is a good enhancement described [in this article](https://ponyfoo.com/articles/serviceworker-messagechannel-postmessage).

Since we establish a global listener on `navigator.serviceWorker`, it can receive messages from other workers that may control the current client. With this check we're proceeding with event handling only if the source of the event (message) is our MSW worker.